File: tests/e2e/runtime/enums/java/postgresql/src/main/java/com/example/db/models/Priority.java
package com.example.db.models;

public enum Priority {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high"),
    CRITICAL("critical");

    private final String value;

    Priority(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static Priority fromValue(String value) {
        for (Priority e : values()) {
            if (e.value.equals(value)) {
                return e;
            }
        }
        throw new IllegalArgumentException("Unknown Priority: " + value);
    }
}

File: tests/e2e/fixtures/enums/postgresql/kotlin/src/main/kotlin/com/example/db/com/example/db/models/Priority.kt
package com.example.db.models

enum class Priority(val value: String) {
    LOW("low"),
    MEDIUM("medium"),
    HIGH("high"),
    CRITICAL("critical");

    companion object {
        fun fromValue(value: String): Priority =
            entries.first { it.value == value }
    }
}
